### DISCUSSION TITLE
[HELP] Cómo puedo configurar un SLA para que me cuente sólo los dias hábiles y no los días festivos o fin de semana?

### DISCUSSION THREAD

**Discussion by NicolasPLG** [Cotalker/documentation#242] [2. Questions and Answers] [answered]
Hola

Quisiera saber cómo se puede configurar un sla que no me tome los días fines de semana o feriados (se que el feriado es más complicado).

Gracias :>

**Comment by guillevalin** [comment, marked as answer]
@NicolasPLG Con la configuración de SLA no podrías realizar esto.

Sin embargo, una manera alternativa sería que tuvieras un schedule configurado de manera semanal (lunes a viernes) el estado de una tarea si cumple las reglas de negocio que tu quieres (para emular el SLA).

Respecto a los feriados, no es una solución trivial, cualquier solución implica cargar una lista de feriados al sistema (sea una rutina, o sea un FaaS).

**Reply by NicolasPLG** [reply]
hola @guillevalin probaré con eso, tengo un tema sobre una regla es que si no responden en 2 días hábiles mande un mensaje. si lo envían un viernes, le mensaje sería el martes, y si es un sábado sería el martes tambien, y no sé si con varios if serviria

**Reply by guillevalin** [reply]
@NicolasPLG podrías hacerlo con una step Custom JavaScript que calcule la diferencia de días hábiles entre dos fechas (createdAt de la task vs la fecha actual) dentro de un schedule que se ejecute diariamente:

https://stackoverflow.com/questions/37069186/calculate-working-days-between-two-dates-in-javascript-excepts-holidays

En ese ejemplo de Stack Overflow hay un ejemplo donde consideran una constante donde se ingresan los feriados, esa variable la podrían estar manteniendo constantemente con los feriados chilenos.

Saludos.

**Reply by NicolasPLG** [reply]
@guillevalin  lo revisaré y lo probaré. Muchas gracias.
